Source/AliveLibAE/MainMenu.cpp

```diff
@@ -2207,6 +2207,13 @@ MainMenuNextCam MainMenuController::LoadDemo_Update_4D1040(u32)
         }
         field_6_flags.Set(Options::eDead_Bit3);
 
+        char_type file[32] = {};
+        sprintf(file, "ATTR%04d.SAV", sDemos_5617F0[demoId].field_A_id);
```

Comment:
I don't see how this is correct when the code below in !gIsDemoStartedManually_5C1B9C changes the value of demoId

Reply:
It changes it for the purpose of having it ready for the next demo

Reply:
To illustrate it better, I didn't move this block of the code, but moved the one that increments sCurrentDemoIdForIdlingDemoPlayback_5C1BA2 under it.

Reply:
But `demoId` is a local, and after `demoId = sCurrentDemoIdForIdlingDemoPlayback_5C1BA2;` its never used again. But it was used before you moved the code hence it seems wrong or needs more refactoring.

Reply:
You're right, demoId isn't used later on. Since I don't see why it should be changed right before loading a savegame (wouldn't make much sense, since it checks if the demo exists earlier on, to pick a different one later), so I removed those assignments. They point to the same variables anyway (albeit not needlessly incremented)

Reply:
But it was used before the change and it doesn't point to the same var - is this fixed based on the original func by comparing with IDA?

Reply:
No, I don't have IDA. The fix was only based on observation that the original game starts demos from the first one forward, and this project from the second one forward